src/internal_events/http_client.rs

```diff
@@ -109,3 +109,29 @@ impl<'a, B: HttpBody> std::fmt::Display for FormatBody<'a, B> {
         }
     }
 }
+
+#[derive(Debug)]
+pub struct HttpBodyDataReceived<'a> {
+    pub buf: &'a mut Vec<u8>,
+}
+
+impl<'a> InternalEvent for HttpBodyDataReceived<'a> {
+    fn emit(self) {
+        // Check if the buffer is empty before any operation
+        if self.buf.is_empty() {
+            return;
+        }
+
```

Comment:
I would not do this check here. Having an empty request or response is actually valid, and should be reported as such.

Reply:
Notably it _could_ be useful debugging information to get an "this request had an empty body" especially if you _don't_ expect it to be empty. I'd just make sure it's obviously emitted, rather than just an ambiguous whitespace or similar 👍